triedb/pathdb/database.go

```diff
@@ -337,6 +337,12 @@ func (db *Database) repairHistory() error {
 	}
 	if pruned != 0 {
 		log.Warn("Truncated extra state histories", "number", pruned)
+		// Reset the state history index metadata to current state id.
+		if meta := loadIndexMetadata(db.diskdb); meta != nil {
+			oldID := meta.Last
+			log.Warn("Reset state history index metadata after truncation", "oldID", oldID, "newID", id)
+			storeIndexMetadata(db.diskdb, id)
```

Comment:
The reset should only be applied if the last indexed item is beyond the state history?

It feels very dangerous to blindly reset the indexing tracker

Reply:
Add a boundary check before reset